docs/router/src/example/building-a-nest.md

```diff
@@ -0,0 +1,263 @@
+# extern crate dioxus;
+# use dioxus::prelude::*;
+#
+#[allow(non_snake_case)]
+fn Blog(cx: Scope) -> Element {
+    cx.render(rsx! {
+        h1 { "Blog" }
+        Outlet {}
+    })
+}
+```
+
+> Note the `Outlet { }` component. For the components of a nested route to be
+> rendered, we need an equally nested outlet.
+
+Now we'll create another index component, that'll be displayed when no blog post
+is selected:
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+#
+#[allow(non_snake_case)]
+fn BlogList(cx: Scope) -> Element {
+    cx.render(rsx! {
+        h2 { "Choose a post" }
+        ul {
+            li {
+                Link {
+                    target: "/blog/1".into(),
+                    "Read the first blog post"
+                }
+            }
+            li {
+                Link {
+                    target: "/blog/2".into(),
+                    "Read the second blog post"
+                }
+            }
+        }
+    })
+}
+```
+
+We also need to create a component that displays an actual blog post. Within
+this component we can use the `use_route` hook to gain access to our URL
+parameters:
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+#
+#[allow(non_snake_case)]
+fn BlogPost(cx: Scope) -> Element {
+    let route = use_route(&cx).unwrap();
+
+    let post_id = route.parameters.get("post_id");
+    let post = post_id
+        .map(|id| id.to_string())
+        .unwrap_or(String::from("unknown"));
+
+    cx.render(rsx! {
+        h2 { "Blog Post: {post}"}
+    })
+}
+```
+
+Finally, let's tell our router about those components.
+```rust,no_run
+# // Hidden lines (like this one) make the documentation tests work.
+# extern crate dioxus;
+# use dioxus::prelude::*;
+# fn Blog(cx: Scope) -> Element { unimplemented!() }
+# fn BlogList(cx: Scope) -> Element { unimplemented!() }
+# fn BlogPost(cx: Scope) -> Element { unimplemented!() }
+# fn Home(cx: Scope) -> Element { unimplemented!() }
+# fn NavBar(cx: Scope) -> Element { unimplemented!() }
+# fn PageNotFound(cx: Scope) -> Element { unimplemented!() }
+#
+#[allow(non_snake_case)]
+fn App(cx: Scope) -> Element {
+    let routes = use_segment(&cx, || {
+        Segment::default()
+            .index(Home as Component)
+            // new stuff starts here
+            .fixed(
+                "blog",
+                Route::new(Blog as Component).nested(
+                    Segment::default().index(BlogList as Component).parameter(
+                        ParameterRoute::new("post_id", BlogPost as Component),
+                    ),
+                ),
+            )
+            // new stuff ends here
+    });
+
+    cx.render(rsx! {
+        Router {
+            routes: routes.clone(),
+            fallback: RcComponent(PageNotFound),
```

Comment:
`fallback()` is part of the `Segment` API.

However, if your blog id (in the example `kajsdhf`) is a parameter route, the router won't render any fallback content. That's because it can't distinguish between a valid blog id and an invalid one.

Reply:
Also, I might have forgotten to mention that method in the guide.

Reply:
Looks to me like `fallback` is a separate prop. I was suggesting adding it as a method on `Segment` (removing this prop).

Then it would be more consistent with the other routes, and also a bit more flexible.

Reply:
It is both, a prop on `Router` and a method on `Segment`. However, you're right, it could and should be more flexible, which would also remove the need for the router prop. I'll do that.

Reply:
Ok, I removed the `fallback` prop from the `Router`. The fallback routes defined on `Segment`s now cascade. Therefore your initial example is now possible (caveat below).

The previous behaviour of the `fallback` prop can simply be recreated by setting a fallback on the root `Segment`.

The aforementioned caveat: in your example you want a different not found page for `/blog`. The fallback will still only work if all your posts are fixed routes. If you use a parameter route and load the post from within your component, that wont work, as the router cannot know that a specific parameter value is invalid.